### PR TITLE
disttask: fix a bug that session may not commit before putting into the session pool

### LINKED ISSUE #44286: session may not commit before putting into the session pool.
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

``` SQL
func execSQL(ctx context.Context, se sessionctx.Context, sql string, args ...interface{}) ([]chunk.Row, error) {
	rs, err := se.(sqlexec.SQLExecutor).ExecuteInternal(ctx, sql, args...)
	if err != nil {
		return nil, err
	}
	if rs != nil {
		rows, err := sqlexec.DrainRecordSet(ctx, rs, 1)
		if err != nil {
			return nil, err
		}
		err = rs.Close()
		if err != nil {
			return nil, err
		}
		return rows, err
	}
	return nil, nil
}
```

It there is any error in DrainRecordSet, rs is not close.

### 4. What is your TiDB version? (Required)
master

### PR DESCRIPTION
**Pull request by wjhuang2016**
### What problem does this PR solve?

Issue Number: close #44286

Problem Summary:
rs.Close() may not call if there is any error in DrainRecordSet.
### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
Start the tiup cluster, kill and restart PD during the test. Add some logs to check if the session is committed. 
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by hawkingrei — APPROVED** [review]

**Review by lance6716 — APPROVED** [review]
I prefer we don't close the issue, just mitigate the severity and add a linter to prevent this problem in future

File: disttask/framework/storage/task_table.go [hunk 1]
```diff
@@ -78,15 +79,8 @@ func execSQL(ctx context.Context, se sessionctx.Context, sql string, args ...int
...
-		if err != nil {
-			return nil, err
-		}
-		err = rs.Close()
-		if err != nil {
-			return nil, err
-		}
-		return rows, err
+		defer terror.Call(rs.Close)
+		return sqlexec.DrainRecordSet(ctx, rs, 1024)
```
Comment (tangenta) [inline disttask/framework/storage/task_table.go:83]: Why change `maxChunkSize` to 1024? I think most internal SQLs should not return so many rows.

Comment (wjhuang2016) [reply disttask/framework/storage/task_table.go:83]: It may get many row when getting all pending subtasks.

**Pull request merged**